Tests/Marketplace/upload_packs.py

```diff
@@ -203,6 +211,44 @@ def upload_index_to_storage(index_folder_path, extract_destination_path, index_b
     print_color(f"Finished uploading {GCPConfig.INDEX_NAME}.zip to storage.", LOG_COLORS.GREEN)
 
 
+def get_private_packs(private_index_path):
+    """
+    Get the list of ID and price of the private packs.
+    Args:
+        private_index_path: The path for the index of the private packs.
+
+    Returns:
+        private_packs: A list of ID and price of the private packs.
+    """
+    metadata_files = glob.glob(f"{private_index_path}/**/metadata.json")
+    private_packs = []
+
+    for metadata_file_path in metadata_files:
+        with open(metadata_file_path, "r") as metadata_file:
+            metadata = json.load(metadata_file)
+        if metadata:
+            private_packs.append({
+                'id': metadata.get('id'),
+                'price': metadata.get('price')
+            })
+
+    return private_packs
+
+
+def add_private_packs_to_index(index_folder_path, private_index_path):
+    """
+    Add the private packs to the index folder.
+    Args:
+        index_folder_path: The index folder path.
+        private_index_path: The path for the index of the private packs.
+
+    """
+    for d in os.scandir(private_index_path):
+        if os.path.isdir(d.path):
+            update_index_folder(index_folder_path, d.name, d.path)
+    shutil.rmtree(private_index_path)
```

Comment:
I think it is better to delete outside of this function, because this is not what this function states it dose.
if you move it, consider wrapping the commands after the creation of `private_index_path` with `try`, and call the deletion with `finally`